ament_cmake_export_interfaces/cmake/ament_cmake_export_interfaces-extras.cmake.in

```diff
@@ -5,7 +5,24 @@ set(_exported_interfaces "@_AMENT_CMAKE_EXPORT_INTERFACES@")
 # include all exported interfaces
 if(NOT _exported_interfaces STREQUAL "")
   foreach(_interface ${_exported_interfaces})
-    include("${@PROJECT_NAME@_DIR}/${_interface}Export.cmake")
-    list(APPEND @PROJECT_NAME@_INTERFACES "@PROJECT_NAME@::${_interface}")
+    set(_export_file "${@PROJECT_NAME@_DIR}/${_interface}Export.cmake")
+    include("${_export_file}")
+
+    # extract the target names associated with the export
+    file(
+      STRINGS "${_export_file}" _foreach_targets
+      REGEX "foreach\\(_expectedTarget (.+)\\)")
+    list(LENGTH _foreach_targets _matches)
+    if(NOT _matches EQUAL 1)
+      message(FATAL_ERROR
+        "Failed to find exported target names in '${_export_file}'")
+    endif()
+    string(LENGTH "${_foreach_targets}" _length)
+    math(EXPR _substring_length "${_length} - 25")
+    string(SUBSTRING "${_foreach_targets}" 24 ${_substring_length} _targets)
```

Comment:
@dirk-thomas I'd be somewhat inclined towards [`STRING(REGEX_REPLACE ...)`](https://cmake.org/cmake/help/v3.12/command/string.html#regex-replace) to get the capture group instead of using clever substrings.

Reply:
Since we already have the regex that sounds like a good idea. The only downside is that we have to evaluate it twice since the matches of the file regex can't be accessed. Done in 50f024e15c121975c789855182f9cbe1d97ffca1.